File: client/src/app/models/account.ts
import { BasicModel } from "./basic-model";

export interface Account extends BasicModel {
    id: string;
    type: string;
    subType: string;
    group: string;
    description: string;
    notes: string;
    code: number;
    sign: number;
    accountsPackage: string;
    createdDate: Date;
    lastModifiedDate: Date;
}

File: server/src/main/java/com/devracom/tyche/msv_chart_of_accounts/Account.java
package com.devracom.tyche.msv_chart_of_accounts;

import com.devracom.tyche.msv_users.dto.RestrictedUser;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Account {
    @Id
    private String id;
    private String type;
    private String subType;
    private String group;
    private String description;
    private String notes;
    private long code;
    private short sign;
    private String accountsPackage;
    private String userId;

    @CreatedDate
    private Date createdDate;

    @LastModifiedDate
    private Date lastModifiedDate;
}
